File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
  // using a class here allows us to have a constructor to init properties when we create it
  gender: string;
  minAge = 18;
  maxAge = 99;
  pageNumber = 1;
  pageSize = 5;
  orderBy = 'lastActive';

  constructor(user: User) {
    // lets not default list to opposite gender
    // this.gender = user.gender === 'female' ? 'male' : 'female'
    this.gender = "all";
  }

}

File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    /* Derives from PaginationParams, so we have all those properties included here */
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }

        public string Gender { get; set; }

        public int MinAge { get; set; } = 18;

        public int MaxAge { get; set; } = 100;
        public string OrderBy { get; set; } = "lastActive";
    }
}